Makefile

```diff
@@ -0,0 +1,101 @@
+TOOLS_DIR="$(PWD)/.tools"
+
+GIT_COMMIT=$(shell git log -1 --format=%H)
+GIT_TAG=$(shell git symbolic-ref -q --short HEAD || git describe --tags --exact-match)
+BUILD_DATE=$(shell date -Ins)
+
+GOLANGCI_LINT_VERSION="v1.63.4"
+PROTOC_GEN_GO_VERSION="v1.36.2"
+PROTOC_GEN_GO_GRPC_VERSION="v1.5.1"
+PROTOC_GEN_GO_GRPC_GATEWAY_VERSION="v2.25.1"
+BUF_VERSION="v1.45.0"
+PROTOC_GEN_OPENAPIV2_VERSION="v2.15.2"
+
+.PHONY: all
+all: agent chorus proxy worker chorctl bench
+
+.PHONY: install-tools
+install-tools:
+	mkdir -p $(TOOLS_DIR)
+	GOBIN=$(TOOLS_DIR) go install golang.org/x/tools/cmd/goimports@latest
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(TOOLS_DIR) $(GOLANGCI_LINT_VERSION)
+	GOBIN=$(TOOLS_DIR) go install "google.golang.org/protobuf/cmd/protoc-gen-go@$(PROTOC_GEN_GO_VERSION)"
+	GOBIN=$(TOOLS_DIR) go install "google.golang.org/grpc/cmd/protoc-gen-go-grpc@$(PROTOC_GEN_GO_GRPC_VERSION)"
+	GOBIN=$(TOOLS_DIR) go install "github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway@$(PROTOC_GEN_GO_GRPC_GATEWAY_VERSION)"
+	GOBIN=$(TOOLS_DIR) go install "github.com/bufbuild/buf/cmd/buf@$(BUF_VERSION)"
+	GOBIN=$(TOOLS_DIR) go install "github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2@$(PROTOC_GEN_OPENAPIV2_VERSION)"
+
+.PHONY: ensure-tools
+ensure-tools:
```

Comment:
will it work with tool version upgrade?

I think that it is better to just run `install-tools` because `go install` already has cache.

Reply:
At the moment it checks only that tools are installed to the directory.

`go install` will not work with `golangci-lint`, as it has different installation method.

Reply:
maybe we can move `golangci-lint` installation to a separate step then?

Reply:
Remade, now version is validated for linter as well.